### PR TITLE
Expand [%src_pos] to hard-coded record

### PR DESCRIPTION
**Pull request by vivianyyd**
Hackily hard-coded a record `position` which contains the same fields as `Lexing.position`.
Also appears to shadow the real `Lexing.position`, as warned by the compiler message "This open statement shadows the label pos_lnum (which is later used)"

### CODE REVIEW

**Review by goldfirere — COMMENTED** [review]
This is definitely working in the right direction. See a few line comments below. Also, as discussed this morning, let's rename `"position"` to `"lexing_position"`. Also, please make the change to `Lexing.position` to make it a synonym of your new `lexing_position`, and add a few tests to make sure that (1) `Lexing.position` and `lexing_position` are the same type and (2) the field names are accessible from within the `Lexing` module.

Thanks!

File: ocaml/testsuite/tests/typing-implicit-source-positions/named_function.ml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+|}]
+
+(* Shadowing *)
+
+type position = int 
+[%%expect{|
+type position = int
+|}]
+
+let _ = f ~src_pos:{pos_fname="hello" ; pos_lnum=1; pos_bol=2; pos_cnum=3} ();;
```
[the remark this answers is not in the document]
Comment (vivianyyd) [reply ocaml/testsuite/tests/typing-implicit-source-positions/named_function.ml]: Pushed new test file - it uses `[%src_pos]` after `lexing_position` is shadowed, which I think tests what we want now

**Review by goldfirere — APPROVED** [review]

**Comment by goldfirere** [comment]
Looks good!

I can't merge this while it's a Draft, but it's ready to merge.

**Comment by vivianyyd** [comment]
Marked as no longer draft

**Pull request merged**